### PR TITLE
fix(flask): fix query string with non-Unicode characters

### PR DESCRIPTION
**Pull request by jd**
This decode the query string using the default Python encoding.

### CODE REVIEW

File: ddtrace/contrib/flask/patch.py [hunk 1]
```diff
@@ -323,7 +322,7 @@ def traced_start_response(status_code, headers):
             config.flask,
             method=request.method,
             url=request.base_url,
-            query=compat.to_unicode(request.query_string),
+            query=request.query_string.decode(errors="backslash"),
```
Comment (majorgreys) [inline ddtrace/contrib/flask/patch.py:325]: Is this meant to be:

```suggestion
            query=request.query_string.decode(errors="backslashreplace"),
```

**Comment by majorgreys** [comment]
Would we rather want the `backslashescape` as the error handling? Not sure if that makes the query string tag salvageable for later use or not but would be more so than just `escape`.

```
>>> u"/?foo=정상처리".encode("euc_kr").decode("utf-8", errors="backslashreplace")
'/?foo=\\xc1\\xa4\\xbb\\xf3ó\\xb8\\xae'
```

**Comment by jd** [comment]
> Would we rather want the backslashescape as the error handling? Not sure if that makes the query string tag salvageable for later use or not but would be more so than just escape.

It does not work with Python 2, so I just went with what works everywhere 🤷

**Review by majorgreys — APPROVED** [review]

File: ddtrace/contrib/flask/patch.py [hunk 2]
```diff
@@ -323,7 +322,9 @@ def traced_start_response(status_code, headers):
             config.flask,
             method=request.method,
             url=request.base_url,
-            query=compat.to_unicode(request.query_string),
+            # It's probably wrong to assume it's UTF-8, but there is no hint to know what the encoding of the query
+            # string is anyhow.
+            query=request.query_string.decode("utf-8", errors="replace"),
```
Comment (brettlangdon) [inline ddtrace/contrib/flask/patch.py:327]: is this a fix we should make in `compat.to_unicode`?

Comment (jd) [reply ddtrace/contrib/flask/patch.py:327]: I'm not sure. I'm not a fan of the `to_unicode` approach which tries to force everything, anyhow, without caring what the input type is, to convert to Unicode.
I'm more into a thoughtful approach on a case-by-case basis.

**Review by P403n1x87 — APPROVED** [review]

**Pull request merged**